### PR TITLE
ports the exosuit pka from tgstation

### PR DESCRIPTION
**Pull request by Erikafox**
## About The Pull Request
ports https://github.com/tgstation/tgstation/pull/60384 from tgstation

## Why It's Good For The Game
I want exosuit pkas for the Vela

## Changelog

:cl:
add: InvalidArgument3: Exosuit-mounted kinetic accelerator equipment. Unlocked through Mining Technology.
/:cl:

### CODE REVIEW

**Comment by Apogee-dev** [comment]
Be sure to add this to the equipment list for mining Ripleys instead of the plasma cutter (or maybe give them a tiny chance for plasma cutter like they have for diamond drill)

**Review by MarkSuckerberg — APPROVED** [review]

File: code/game/mecha/equipment/weapons/weapons.dm [hunk 1]
```diff
@@ -151,6 +151,25 @@
...
+	fire_sound = 'sound/weapons/kenetic_accel.ogg'
+	harmful = TRUE
+
+//attachable to all mechas, like the plasma cutter
+/obj/item/mecha_parts/mecha_equipment/weapon/energy/mecha_kineticgun/can_attach(obj/mecha/working/M)
+	if(..()) //combat mech
+		return 1
+	else if(M.equipment.len < M.max_equip && istype(M))
+		return 1
+	return 0
```
Comment (Fikou) [inline code/game/mecha/equipment/weapons/weapons.dm]: use return TRUE and return FALSE

Comment (Fikou) [reply code/game/mecha/equipment/weapons/weapons.dm]: and a descriptive variable name, not M

Comment (Erikafox) [reply code/game/mecha/equipment/weapons/weapons.dm]: I think the M goes back way further than right there.

Comment (Ms-Mee) [reply code/game/mecha/equipment/weapons/weapons.dm]: https://github.com/shiptest-ss13/Shiptest/blob/master/.github/CONTRIBUTING.md#use-self-explanatory-var-names

Comment (Ms-Mee) [reply code/game/mecha/equipment/weapons/weapons.dm]: From what I can tell even with ports the ideal is to follow our contribution guidelines to have decently readable and modifiable code, including avoiding single character variable naming even when it seems pointless or is the standard elsewhere

Comment (Erikafox) [reply code/game/mecha/equipment/weapons/weapons.dm]: alright

**Pull request merged**